modules/bam2fastq/1.1/bam2fastq.smk

```diff
@@ -33,35 +33,46 @@ localrules:
 ##### RULES #####
 
 
-# Symlinks the input files into the module results directory (under '00-inputs/')
+def get_genome_build(wildcards):
+    tbl = config["lcr-modules"]["bam2fastq"]["samples"]
+    genome_build = tbl[(tbl.sample_id == wildcards.sample_id) & (tbl.seq_type == wildcards.seq_type)]["genome_build"]
+    return reference_files(expand("genomes/{genome_build}/genome_fasta/genome.fa", genome_build = genome_build))
+
+def get_bams(wildcards,build = False):
+    tbl = config["lcr-modules"]["bam2fastq"]["samples"]
+    return(expand("data/{{seq_type}}_bams/{{sample_id}}.{genome_build}.bam",genome_build = tbl[(tbl.sample_id == wildcards.sample_id) & (tbl.seq_type == wildcards.seq_type)]["genome_build"]))
+
+
 rule _bam2fastq_input_bam:
     input:
-        bam = CFG["inputs"]["sample_bam"]
+        #create a symlink here that does not require the {genome_build}
+        bam_path = get_bams
     output:
-        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam"
+        bam = CFG["dirs"]["inputs"] + "{seq_type}/{sample_id}.bam"
     run:
-        op.relative_symlink(input.bam, output.bam)
+        op.relative_symlink(input, output.bam)
 
 
 # Conditional rules depending on whether or not fastq outputs will be temporary
 if CFG["temp_outputs"] == True:
     rule _bam2fastq_run:
         input:
             bam = str(rules._bam2fastq_input_bam.output.bam),
-            genome = reference_files("genomes/{genome_build}/genome_fasta/genome.fa")
```

Comment:
I'm struggling understanding the rationale or need for this change

Reply:
Is it somehow a way to avoid having genome-build end up in the output file names?